### DISCUSSION TITLE
Quick Question on Margin and/or Padding with Text within an shape

### DISCUSSION THREAD

**Discussion by kibblewhite** [excalidraw/excalidraw#6065] [Q&A]
Hey team,

I was curious if there is a way to put a margin and/or padding within a shape. So basically, I draw a shape, I double click to add some text, and now I would like some padding or a margin (depending on how you view it) on the text so there is some spacing between the edges of the shape and the text. I think this becomes more apparent when you align your text in any other way than just middle-centred.

I think I must be missing something, as this is normally a simple step, but I can not for the life of me find how to do this. Is it something to do with styles, and if so how do I go about changing this? Any help would be greatly appreciated...

**Comment by pmz** [comment]
Curious why this is still unanswered. I love Excalidraw and this is a big problem when using it to brainstorm.

**Comment by gsans** [comment]
While there's no automatic way of doing this you can: 

1) Create shape and text separately.
2) Select both and align to center both vertically and horizontally.
3) Group both shape and text. 
4) Copy/paste the group to use again as a "template". 
5) Change colours and text by double-clicking (text, shape).
6) Ungroup if you need to re-touch or change size for text.
7) Group again to restore the "template".
